ethstorage/encoder/optimism_encoder.go

```diff
@@ -0,0 +1,224 @@
+package encoder
```

Comment:
I would suggest not putting those codes into the encoder folder because it is for the storage encoding, not for the blob encoding. They should be together with "utils.DecodeBlob"